[all hosts] (Auth) Add support table for NAA preview (#4698)
OfficeDev/office-js-docs-pr: docs/develop/enable-nested-app-authentication-in-your-add-in.md

- You can use incremental and dynamic consent for scopes.
- You don't need to [preauthorize your hosts](/microsoftteams/platform/m365-apps/extend-m365-teams-personal-tab?tabs=manifest-teams-toolkit#update-azure-ad-app-registration-for-sso) (For example, Teams, Office) to call your endpoints.

## NAA supported accounts and hosts

> [!IMPORTANT]
> Nested app authentication (NAA) is currently in preview. To try this feature, join the Microsoft 365 Insider Program (https://insider.microsoft365.com/join) and choose the Beta Channel. Don't use NAA in production add-ins. We invite you to try out NAA in test or development environments and welcome feedback on your experience through GitHub (see the **Feedback** section at the end of this page).

> NAA supports both Microsoft Accounts and Microsoft Entra ID (work/school) identities. It doesn’t support Azure AD B2C for business-to-consumer identity management scenarios. The following table explains the current support in preview. Anything listed as "Coming soon" will be supported by the time that NAA is made generally available.

Reviewer: Isn't Azure AD a deprecated term?

Author: Doesn't seem that way.
https://learn.microsoft.com/en-us/azure/active-directory-b2c/overview

> NAA supports both Microsoft Accounts and Microsoft Entra ID (work/school) identities. It doesn't support Azure AD B2C for business-to-consumer identity management scenarios. The following table explains the current support in preview. Anything listed as "Coming soon" will be supported by the time that NAA is made generally available.

Reviewer: ```suggestion
NAA supports both Microsoft Accounts and Microsoft Entra ID (work/school) identities. It doesn't support Azure Active Directory B2C for business-to-consumer identity management scenarios. The following table explains the current support in preview. Anything listed as "Coming soon" will be supported by the time that NAA is made generally available.
```

- You can use incremental and dynamic consent for scopes.

> - You don't need to [preauthorize your hosts](/microsoftteams/platform/m365-apps/extend-m365-teams-personal-tab?tabs=manifest-teams-toolkit#update-azure-ad-app-registration-for-sso) (For example, Teams, Office) to call your endpoints.

Reviewer: ```suggestion
- You don't need to [preauthorize your hosts](/microsoftteams/platform/m365-apps/extend-m365-teams-personal-tab?tabs=manifest-teams-toolkit#update-azure-ad-app-registration-for-sso) (for example, Teams, Office) to call your endpoints.
```

NAA supports both Microsoft Accounts and Microsoft Entra ID (work/school) identities. It doesn't support Azure Active Directory B2C for business-to-consumer identity management scenarios. The following table explains the current support in preview. Anything listed as "Coming soon" will be supported by the time that NAA is made generally available.

| Application | Windows                                                                            | Mac | Web          | iOS/iPad   | Android        |
|-------------|------------------------------------------------------------------------------------|-----|--------------|------------|----------------|
| Excel       | Yes (Current Channel (Preview))                                                    | Yes | Coming soon  | Yes (iPad) | Not applicable |

> | Outlook     | Yes* (Current Channel (Preview) for classic Outlook only, new Outlook coming soon) | Yes | Yes          | Yes (iOS)  | Not applicable |

Reviewer: Originally, Android was "Yes" on Outlook. Just checking if it should still say that.

```suggestion
| Outlook     | Yes* (Current Channel (Preview) for classic Outlook only, new Outlook coming soon) | Yes | Yes          | Yes (iOS)  | Yes            |
```

## NAA supported accounts and hosts

> NAA supports both Microsoft Accounts and Microsoft Entra ID (work/school) identities. It doesn’t support B2C scenarios. The following table explains the current support in preview. Anything listed as "Coming soon" will be supported by the time that NAA is made generally available.

Reviewer 2: What's "B2C"?

Author: Changed to "It doesn't support Azure AD B2C for business-to-consumer identity management scenarios."

> NAA supports both Microsoft Accounts and Microsoft Entra ID (work/school) identities. It doesn’t support B2C scenarios. The following table explains the current support in preview. Anything listed as "Coming soon" will be supported by the time that NAA is made generally available.

Reviewer: ```suggestion
NAA supports both Microsoft Accounts and Microsoft Entra ID (work/school) identities. It doesn't support B2C scenarios. The following table explains the current support in preview. Anything listed as "Coming soon" will be supported by the time that NAA is made generally available.
```

> | Application | Windows                                                               | Mac | Web          | iOS | Android |

Reviewer: Consider indicating iPad vs iOS support (e.g., "iPad" or "Yes (iPad)" for WXP, "iOS" or "Yes (iOS)" for Outlook)

```suggestion
| Application | Windows                                                               | Mac | Web          | iOS/iPad | Android |
```

|-------------|-----------------------------------------------------------------------|-----|--------------|-----|---------|

> | Excel       | Yes (Insiders Slow)                                                   | Yes | Coming soon  | Yes | No      |

Reviewer 3: new name for this channel is Current Channel (Preview)

Author: Thanks! What does Insiders Fast map to? "Beta Channel"?

> | Excel       | Yes (Insiders Slow)                                                   | Yes | Coming soon  | Yes | No      |
> | Outlook     | Yes (Insiders Slow for classic Outlook only, new Outlook coming soon) | Yes | Yes          | Yes | Yes     |
> | PowerPoint  | Yes (Insiders Slow)                                                   | Yes | Coming soon  | Yes | No      |
> | Word        | Yes (Insiders Slow)                                                   | Yes | Coming soon  | Yes | No      |

Reviewer: ```suggestion
| Excel       | Yes (Insiders Slow)                                                   | Yes | Coming soon  | Yes | Not applicable      |
| Outlook     | Yes (Insiders Slow for classic Outlook only, new Outlook coming soon) | Yes | Yes          | Yes | Yes     |
| PowerPoint  | Yes (Insiders Slow)                                                   | Yes | Coming soon  | Yes | Not applicable      |
| Word        | Yes (Insiders Slow)                                                   | Yes | Coming soon  | Yes | Not applicable      |
```
